composer/src/main/java/com/rewedigital/examples/msintegration/composer/composing/parser/Composition.java

```diff
@@ -0,0 +1,34 @@
+package com.rewedigital.examples.msintegration.composer.composing.parser;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+
+public class Composition {
```

Comment:
Idee: Jedes Template (main oder content) wird abgebildet auf eine Composition. Diese enthält (rekursiv) die Compositions der inkludierten Templates auf folgende Weise: Für jeden Belang des Kompositionsprozesses (Body, CSS) gibt es einen OngoingComposition.Handler, welcher der Composition einen Composition.Part hinzufügt. Dabei aggregiert der Handler "seine" Parts der inkludierten Parts zu einem Part, welcher dem Ergebnis der OngoingComposition hinzugefügt wird:
- Der Body-Handler z.B. ersetzt inklude-Tags im Template mit den Inhalten der inkludierten Templates (D.h. mit den contents der entsprechenden Parts der zu den inkludierten Templates gehörenden Compositions).
- Der CSS-Handler sammelt die Asset-Links der inkludierten Templates in einer Liste und dedupliziert diese.

Eine Composition kann über Composition#body() in einen Response-Body (d.h. in das fertig komponierte HTML Dokument) transformiert werden. Dazu bekommen alle Composition.Parts die Möglichkeit, den Body zu erweitern (Part#enrichBody(String body)). 
Oder die Composition wird in einer übergeordneten Composition verwendet, dann können OngoingComposition.Handlers für die Composition#find() Methode den/die Parts finden, die sie aggregieren wollen.